### PR TITLE
feat(node): add package.json analyzer

### PR DESCRIPTION
**Pull request by ankk13**
go-dep-parser is updated to consider package.json. Updated respective required files.

### CODE REVIEW

**Comment by masahiro331** [comment]
Please add tests.

File: analyzer/library/npm/npm.go [hunk 1]
```diff
@@ -19,12 +20,18 @@ func init() {
...
-var requiredFiles = []string{"package-lock.json"}
+var requiredFiles = []string{"package-lock.json", "package.json"}
 
 type npmLibraryAnalyzer struct{}
 
-func (a npmLibraryAnalyzer) Analyze(target analyzer.AnalysisTarget) (*analyzer.AnalysisResult, error) {
-	res, err := library.Analyze(types.Npm, target.FilePath, target.Content, npm.Parse)
+func (a npmLibraryAnalyzer) Analyze(target analyzer.AnalysisTarget) (res *analyzer.AnalysisResult, err error) {
+	switch target.FilePath {
+	case "package-lock.json":
```
Comment (masahiro331) [inline analyzer/library/npm/npm.go:29]: target.FilePath is not filename. 
Please use strings.HasSuffix.

File: analyzer/library/npm/npm.go [hunk 2]
```diff
@@ -19,12 +20,18 @@ func init() {
 
 const version = 1
```
Comment (masahiro331) [inline analyzer/library/npm/npm.go:21]: Please change the version. 
https://github.com/aquasecurity/fanal/pull/156

File: analyzer/library/npm/npm.go [hunk 3]
```diff
@@ -19,12 +20,18 @@ func init() {
...
 
-var requiredFiles = []string{"package-lock.json"}
+var requiredFiles = []string{"package-lock.json", "package.json"}
```
Comment (masahiro331) [inline analyzer/library/npm/npm.go:23]: IMO: If you want to use meaningful strings in multiple places, it is preferable to define them const.

File: analyzer/library/npm/npm.go [hunk 4]
```diff
@@ -19,12 +20,18 @@ func init() {
...
+func (a npmLibraryAnalyzer) Analyze(target analyzer.AnalysisTarget) (res *analyzer.AnalysisResult, err error) {
+	switch target.FilePath {
+	case "package-lock.json":
+		res, err = library.Analyze(types.Npm, target.FilePath, target.Content, npm.Parse)
+	case "package.json":
+		res, err = library.Analyze(types.Npm, target.FilePath, target.Content, packagejson.Parse)
+	}
+
 	if err != nil {
 		return nil, xerrors.Errorf("unable to parse package-lock.json: %w", err)
```
Comment (masahiro331) [inline analyzer/library/npm/npm.go:36]: Please return it separately.
Always display package-lock.json error.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**